components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/ExportUtils.java

```diff
@@ -0,0 +1,1087 @@
+                        metadataDTO.getAlias() + ".crt");
+                certificatesList.add(modifiedCertificateMetadata);
+            } catch (APIManagementException e) {
+                log.error("Error retrieving certificate content. For tenantId: " + tenantId + " hostname: "
+                        + url + " alias: " + metadataDTO.getAlias(), e);
+            } catch (IOException e) {
+                log.error("Error while converting certificate content to Byte Array. For tenantId: " + tenantId
+                        + " hostname: " + url + " alias: " + metadataDTO.getAlias(), e);
+            } catch (APIImportExportException e) {
+                log.error("Error while writing the certificate content. For tenantId: " + tenantId + " hostname: "
+                        + url + " alias: " + metadataDTO.getAlias(), e);
+            } finally {
+                if (certificate != null) {
+                    IOUtils.closeQuietly(certificate);
+                }
+            }
+        });
+        return certificatesList;
+    }
+
+    /**
+     * Get endpoint url list from endpoint config.
+     *
+     * @param endpointConfig JSON converted endpoint config
+     * @param type           End point type - production/sandbox
+     * @return List of host names
+     */
+    private static List<String> getEndpointURLs(JSONObject endpointConfig, String type, String apiName) {
+        List<String> urls = new ArrayList<>();
+        if (endpointConfig != null) {
+            try {
+                Object item;
+                item = endpointConfig.get(type);
+                if (item instanceof JSONArray) {
+                    JSONArray endpointsJSON = new JSONArray(endpointConfig.getJSONArray(type).toString());
+                    for (int i = 0; i < endpointsJSON.length(); i++) {
+                        try {
+                            String urlValue = endpointsJSON.getJSONObject(i).get(APIConstants.API_DATA_URL).toString();
+                            urls.add(urlValue);
+                        } catch (JSONException ex) {
+                            log.error("Endpoint URL extraction from endpoints JSON object failed in API: "
+                                    + apiName, ex);
+                        }
+                    }
+                } else if (item instanceof JSONObject) {
+                    JSONObject endpointJSON = new JSONObject(endpointConfig.getJSONObject(type).toString());
+                    try {
+                        String urlValue = endpointJSON.get(APIConstants.API_DATA_URL).toString();
+                        urls.add(urlValue);
+                    } catch (JSONException ex) {
+                        log.error("Endpoint URL extraction from endpoint JSON object failed in API: " + apiName, ex);
+                    }
+                }
+            } catch (JSONException ex) {
+                log.info("Endpoint type: " + type + " not found in API: " + apiName);
```

Comment:
Should this scenario be ignored?

Reply:
No, this should be thrown as an exception. Will fix it. Thanks!

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9406/commits/150d9abd9d4eb25c7f1217fc53f6143f94167542